abe/abe_test.go

```diff
@@ -0,0 +1,158 @@
+package abe
```

Comment:
Scheme tests should be in a separate package, e.g. `abe_test`. This is to make sure we test the API as a black box (the same way as the user will see it). Here, this would prevent an issue with `newAbe` being unexported (you would not have been able to call it from another package).

Reply:
Thanks for the comment. Due to this comment I have:
- split the tests into two part abe_test.go and abe_internal_test.go with package abe_test and abe, respectively
- made parameters of ABE struct public, since user might need to use the prime number p or check l to se how many attributes are allowed
- made parameters of the MSP struct public, since the user might want to change them, add a new attribute...